edx_rest_framework_extensions/tests/test_permissions.py

```diff
@@ -221,78 +223,140 @@ def test_has_permission(self, user_filters, requested_username, expected_result)
+        self.assertIn(expected_log.text, mock_log_method.call_args_list[0][0][0])
+        if expected_log.param is not None:
+            self.assertEqual(mock_log_method.call_args_list[0][0][1], expected_log.param)
 
     def test_anonymous_fails(self):
         request = self._create_request()
         response = self.SomeClassView().dispatch(request)
-        self.assertEqual(response.status_code, 403)
+        self.assertEqual(response.status_code, 401)
 
     def test_session_staff_succeeds(self):
-        request = self._create_request(is_staff=True)
-        self._create_session(request)
+        user = self._create_user(is_staff=True)
+        request = self._create_request()
+        self._create_session(request, user)
 
         response = self.SomeClassView().dispatch(request)
         self.assertEqual(response.status_code, 200)
 
     @patch('edx_rest_framework_extensions.permissions.log')
     def test_session_user_not_in_url_fails(self, mock_log):
+        user = self._create_user()
         request = self._create_request()
-        self._create_session(request)
+        self._create_session(request, user)
         
         response = self.SomeClassView().dispatch(request)
         self.assertEqual(response.status_code, 403)
-        self._assert_in_log('IsUserInUrl', mock_log.info)
+        self._assert_log(mock_log, self.IsUserInUrlLog)
 
     def test_session_user_in_url_succeeds(self):
-        request = self._create_request(username_in_url=True)
-        self._create_session(request)
+        user = self._create_user()
+        request = self._create_request(username_in_url=user.username)
+        self._create_session(request, user)
 
         response = self.SomeClassView().dispatch(request)
         self.assertEqual(response.status_code, 200)
 
+    JwtTestCase = namedtuple('JwtTestCase', 'is_enforced, is_restricted, is_user_in_url, expected_response, expected_log')
+
     @patch('edx_rest_framework_extensions.permissions.waffle.switch_is_active')
-    def test_jwt_unrestricted_succeeds(self, waffle_mock):
-        waffle_mock.return_value = True
-        request = self._create_request(username_in_url=True)
-        self._create_jwt(request)
+    @patch('edx_rest_framework_extensions.permissions.log')
+    @ddt.data(
+        # **** Enforced ****
+        # unrestricted
+        JwtTestCase(is_enforced=True, is_restricted=False, is_user_in_url=True, expected_response=200, expected_log=JwtIsUnrestrictedLog),
+        JwtTestCase(is_enforced=True, is_restricted=False, is_user_in_url=False, expected_response=403, expected_log=IsUserInUrlLog),
+
+        # restricted
+        JwtTestCase(is_enforced=True, is_restricted=True, is_user_in_url=True, expected_response=403, expected_log=JwtHasScopeLog),
+        JwtTestCase(is_enforced=True, is_restricted=True, is_user_in_url=False, expected_response=403, expected_log=JwtHasScopeLog),
+
+        # **** Unenforced ****
```

Comment:
Nit: Easier on the mind if Unenforced came first, to match the order of unrestricted/restricted.

Reply:
do this.

Reply:
updated.